yaml.go

```diff
@@ -389,6 +389,21 @@ type Node struct {
 	// These fields are not respected when encoding the node.
 	Line   int
 	Column int
+
+	// Index holds the node's start position in the decoded YAML text as a
+	// zero-based offset. This fields is not respected when encoding the node.
+	Index int
```

Comment:
should we be explicit here that this is a run offset and not a byte offset?

Reply:
Maybe. The doc strings for new fields mostly just follow the style of the existing Line/Column doc string so those would also need to be updated.